content/filterx/filterx-comparing/_index.md

```diff
@@ -0,0 +1,135 @@
+
+```shell
+filterx f_pid { ${HOST} + string(${PID}) == ${HOST}; };
+```
+
+The following expression selects log messages that do not contain a PID. Also, it uses a template as the left argument of the operator and compares the values as strings:
+
+```shell
+filterx f_pid {"${HOST}${PID}" == ${HOST} };
+```
+
+The following example selects messages with priority level higher than 5.
+
+```shell
+filterx f_level {
+    ${LEVEL_NUM} > 5;
+};
+```
+
+<!-- FIXME more filterx-specific examples? -->
+
+Make sure to:
+
+- Enclose strings and templates in double-quotes.
+- Use the `$` character before macros.
+
+Note that:
+
+- You can use type casting anywhere where you can use templates to apply a type to the result of the template expansion.
+- Using comparator operators can be equivalent to using filter functions, but is somewhat slower. For example, using `"${HOST}" == "myhost"` is equivalent to using `host("myhost" type(string))`.
+- You can use any macro in the expression, including user-defined macros from parsers and classifications.
+- The results of filter functions are boolean values, so they cannot be compared to other values.
+- You can use boolean operators to combine comparison expressions.
+
+## Compare the type
+
+To compare the values of operands and verify that they have the same type, use the `===` operator. The following example defines a string variable with the value "5" as string and uses it in different comparisons:
+
+```shell
+mystring = "5"; # Type is string
+mystring === 5; # false, because the right-side is an integer
+mystring === "5"; # true
+};
+```
+
+To compare only the type, you can use the [`istype` function]({{< relref "/filterx/function-reference.md#istype" >}}).
+<!-- FIXME examples -->
+
+## Strict inequality operator
+
+Compares the values of operands and returns `true` if they are different. Also returns `true` if the value of the operands are the same, but their type is different. For example:
+
+```shell
+"example" !== "example"; # False, because they are the same and both are strings
+"1" !== 1; # True, because one is a string and the other an integer
+```
+
+## Comparison operators
+
+The following numerical and string comparison operators are available.
+
+| Numerical or string operator | String operator | Meaning               |
+| ------------------ | --------------- | --------------------- |
+| ==                | eq              | Equals                |
+| !=                | ne              | Not equal to          |
+| >                 | gt              | Greater than          |
+| <                 | lt              | Less than             |
+| >=                | ge              | Greater than or equal |
+| =<                | le              | Less than or equal    |
+| ===               |                 | Equals and has the same type |
+| !==               |                 | Not equal to or has different type |
```

Comment:
The string operators are introduced here, and I don't know if it is understandable for the reader just by this name. I think it is worth a smaller section.